framework/src/base/MooseApp.C

```diff
@@ -963,7 +967,14 @@ MooseApp::setupOptions()
     else
       _parser.parse(_input_filenames, getParam<std::string>("_input_text"));
 
-    if (isParamValid("mesh_only"))
+    if (isParamValid("metadata_only"))
+    {
+      _action_warehouse.setFinalTask("create_added_mesh_generators");
+      if (isParamValid("mesh_only"))
+        mooseWarning(
```

Comment:
```suggestion
        mooseError(
```

Reply:
@loganharbour please resolve this comment once it is addressed in #26296